redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/flexbox/FlexboxEngine.kt

```diff
@@ -67,59 +67,27 @@ public class FlexboxEngine {
    * Returns the nodes contained in the flexbox.
    */
   public val nodes: MutableList<FlexNode> = ObservableMutableList(
-    onChange = { indexToReorderedIndex = null },
+    onChange = { reorderedNodes = listOf() },
   )
 
   /**
-   * The computed flex lines after calling [applyMeasure].
+   * Holds the list of [nodes] after [FlexNode.order] has been taken into account.
    */
-  internal var flexLines = listOf<FlexLine>()
-
-  /**
-   * Holds the reordered indices after [FlexNode.order] has been taken into account.
-   */
-  private var indexToReorderedIndex: IntArray? = null
-
-  /**
-   * Holds the 'frozen' state of children during measure. If a node is frozen it will no longer
-   * expand or shrink regardless of flex grow/flex shrink attributes.
-   */
-  private var childrenFrozen: BooleanArray? = null
-
-  /**
-   * Map the node index to the flex line which contains the node represented by the index to
-   * look for a flex line from a given node index in a constant time.
-   * Key: index of the node
-   * Value: index of the flex line that contains the given node
-   *
-   * E.g. if we have following flex lines,
-   *
-   * FlexLine(0): itemCount 3
-   * FlexLine(1): itemCount 2
-   *
-   * this instance should have following entries
-   *
-   * [0, 0, 0, 1, 1, ...]
-   */
-  internal var indexToFlexLine: IntArray? = null
-
-  /**
-   * Cache the measured spec. The first 32 bit represents the height measure spec, the last
-   * 32 bit represents the width measure spec of each flex item.
-   * E.g. an entry is created like `(long) heightMeasureSpec << 32 | widthMeasureSpec`
-   *
-   * To retrieve a widthMeasureSpec, call [unpackLower] or [unpackHigher] for a heightMeasureSpec.
-   */
-  private var measureSpecCache: LongArray? = null
+  private var reorderedNodes: List<FlexNode> = listOf()
+    get() {
+      if (field.size != nodes.size) {
```

Comment:
What if I remove and add an item between calls to this function?

Reply:
The field is reset to an empty list every time `nodes` changes: https://github.com/cashapp/redwood/pull/406/files#diff-df8627befd69b95c6ee8f3b8240fd22ea54df1060b9fbf8c167bc5a4360c4ff1L70